internal/ansible/ansible.go

```diff
@@ -53,28 +37,32 @@ func RunPlaybook(id string, playbook []byte, correlationID string) (chan Event,
 	}
 	_ = status.Close()
 
-	events := make(chan Event)
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
A race condition is possible (I encountered it while developing) where the
channel is closed before the last event message is read from the disk and sent
to the channel. The `started` goroutine creates two additional goroutines that
receive inotify events; those routines process the incoming events and
eventually send them to the `events chan json.RawMessage` channel. Because this
channel is closed in the callback handler to `WaitProcess`, when the
`ansible-playbook` process exits, the channel gets closed. Sometimes the inotify
goroutines are still reading and marshaling the files from the artifacts
directory after `ansible-playbook` exits. When that happens, the inotify
goroutines end up trying to send messages to a closed channel. Guarding the
channel with a waitgroup is a simple solution to ensure that all the goroutines
that send to it are done before closing the channel.